<launch>
	<arg name="sub_msg_to_robot_topic_name"       default="/hsrb/message/to_robot" />
	<arg name="pub_msg_to_human_topic_name"       default="/hsrb/message/to_human" />
	<arg name="sub_joint_state_topic_name"        default="/hsrb/joint_states" />
	<arg name="pub_base_twist_topic_name"         default="/hsrb/command_velocity" />
	<arg name="pub_arm_trajectory_topic_name"     default="/hsrb/arm_trajectory_controller/command" />
	<arg name="pub_gripper_trajectory_topic_name" default="/hsrb/gripper_trajectory_controller/command" />
	
	<arg name="sigverse_ros_bridge_port"        default="50001" />
	<arg name="ros_bridge_port"                 default="9090" />

	<node name="hsr_teleop_key" pkg="sigverse_hsr_teleop_key" type="hsr_teleop_key" output="screen" launch-prefix="xterm -e">

		<param name="sub_msg_to_robot_topic_name"       type="string" value="$(arg sub_msg_to_robot_topic_name)" />
		<param name="pub_msg_to_human_topic_name"       type="string" value="$(arg pub_msg_to_human_topic_name)" />
		<param name="sub_joint_state_topic_name"        type="string" value="$(arg sub_joint_state_topic_name)" />
		<param name="pub_base_twist_topic_name"         type="string" value="$(arg pub_base_twist_topic_name)" />
		<param name="pub_arm_trajectory_topic_name"     type="string" value="$(arg pub_arm_trajectory_topic_name)" />
		<param name="pub_gripper_trajectory_topic_name" type="string" value="$(arg pub_gripper_trajectory_topic_name)" />
	</node>

	<group ns="sigverse_ros_bridge">
		<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port)"/>
	</group>
	
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" > 
		<arg name="port" value="$(arg ros_bridge_port)"/>
	</include>
</launch>

